src/makeCalculator.js

```diff
@@ -37,7 +37,47 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    operate: (callback, num) => {
+      callback(num);
+
+      return calculator;
+    },
+
+    reset: () => {
+      calculator.result = 0;
+
+      return calculator;
+    },
+
+    add: (num) => {
+      calculator.result += num;
+
+      return calculator;
+    },
+
+    subtract: (num) => {
+      calculator.result -= num;
+
+      return calculator;
+    },
+
+    multiply: (num) => {
+      calculator.result *= num;
+
+      return calculator;
+    },
+
+    divide: (num) => {
+      calculator.result /= num;
```

Comment:
There is no check for division by zero. This will lead to Infinity result or even crash.